### PR TITLE
Supports long press on URLs inside of an email

### PR DESCRIPTION
**Pull request by LunarX**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 1]
```diff
@@ -484,11 +490,11 @@ class ThreadFragment : Fragment() {
         toolbarSubject.text = subject
 
         threadSubject.setOnLongClickListener {
-            copyStringToClipboard(subject, R.string.snackbarSubjectCopiedToClipboard, mainViewModel.snackBarManager)
+            context.copyStringToClipboard(subject, R.string.snackbarSubjectCopiedToClipboard, mainViewModel.snackBarManager)
             true
         }
         toolbarSubject.setOnLongClickListener {
-            copyStringToClipboard(subject, R.string.snackbarSubjectCopiedToClipboard, mainViewModel.snackBarManager)
+            context.copyStringToClipboard(subject, R.string.snackbarSubjectCopiedToClipboard, mainViewModel.snackBarManager)
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:497]:
```suggestion
            requireContext().copyStringToClipboard(subject, R.string.snackbarSubjectCopiedToClipboard, mainViewModel.snackBarManager)
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 2]
```diff
@@ -484,11 +490,11 @@ class ThreadFragment : Fragment() {
         toolbarSubject.text = subject
 
         threadSubject.setOnLongClickListener {
-            copyStringToClipboard(subject, R.string.snackbarSubjectCopiedToClipboard, mainViewModel.snackBarManager)
+            context.copyStringToClipboard(subject, R.string.snackbarSubjectCopiedToClipboard, mainViewModel.snackBarManager)
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:493]:
```suggestion
            requireContext().copyStringToClipboard(subject, R.string.snackbarSubjectCopiedToClipboard, mainViewModel.snackBarManager)
```

**Review by tevincent — CHANGES_REQUESTED** [review]
Improvements

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 1 above)
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:497]: `context` is not the one from the Fragment but the one from binding thanks to our extensions, yeah it's can be confusing at first

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 2 above)
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:493]: Same answer as previous one

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 3]
```diff
@@ -311,6 +316,7 @@ class ThreadFragment : Fragment() {
             navigateToNewMessageActivity = { uri ->
                 safeNavigateToNewMessageActivity(NewMessageActivityArgs(mailToUri = uri).toBundle())
             },
+            promptLink = { url -> linkContextualMenuAlertDialog.show(url) }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:319]:
```suggestion
            promptLink = linkContextualMenuAlertDialog::show
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt [hunk 4]
```diff
@@ -439,6 +444,33 @@ class ThreadAdapter(
...
     }
 
+    private fun WebView.setupLinkContextualMenu(onClicked: (String) -> Unit) {
+        setOnLongClickListener {
+            val result = hitTestResult
+
+            when (result.type) {
+                WebView.HitTestResult.SRC_IMAGE_ANCHOR_TYPE,
+                WebView.HitTestResult.SRC_ANCHOR_TYPE -> {
+                    getLinkFromResult(result)?.let { url -> onClicked(url) }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt:454]:
```suggestion
                    getLinkFromResult(result)?.let(onClicked)
```
This needs to be checked, I'm not 100% sure it's the right way to write this.

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**